receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -286,14 +286,26 @@ func (ma *MetricsAdjuster) isReset(metricType metricspb.MetricDescriptor_Type,
 		// note: sum of squared deviation not currently supported
 		currentDist := current.GetDistributionValue()
 		previousDist := previous.GetDistributionValue()
+		if currentDist == nil || previousDist == nil {
+			return false
+		}
 		if currentDist.Count < previousDist.Count || currentDist.Sum < previousDist.Sum {
 			// reset detected
 			return false
 		}
 	case metricspb.MetricDescriptor_SUMMARY:
 		currentSummary := current.GetSummaryValue()
 		previousSummary := previous.GetSummaryValue()
-		if currentSummary.Count.GetValue() < previousSummary.Count.GetValue() || currentSummary.Sum.GetValue() < previousSummary.Sum.GetValue() {
+		if currentSummary == nil || previousSummary == nil {
+			return false
+		}
+		if (currentSummary.Count != nil &&
+			previousSummary.Count != nil &&
+			currentSummary.Count.GetValue() < previousSummary.Count.GetValue()) ||
+
+			(currentSummary.Sum != nil &&
+				previousSummary.Sum != nil &&
+				currentSummary.Sum.GetValue() < previousSummary.Sum.GetValue()) {
 			// reset detected
 			return false
 		}
```

Comment:
We can make this much cleaner by a reverse conditional
```go
if currentSummary.Count == nil || previousSummary.Count == nil {
   return true
}
if currentSummary.Count.GetValue() < previousSummary.Count.GetValue() {
   return false
}
if currentSummary.Sum == nil || previousSummary.Sum == nil {
   return true
}
return currentSummary.Sum.GetValue() >= previousSummary.Sum.GetValue()
```

Reply:
I'm not sure that is functionally equivalent.  It seems that the count and sum attributes on summary datapoints are both optional.  We should still detect resets when only one of them is present.

Reply:
Please tailor it to the equivalent, but the point still stands we can avoid that overly involved single line comparison.